pkg/kubelet/kubelet.go

```diff
@@ -547,25 +550,42 @@ func NewMainKubelet(kubeCfg *componentconfig.KubeletConfiguration, kubeDeps *Kub
 
 		switch kubeCfg.ContainerRuntime {
 		case "containerd":
-			cs := containerdshim.NewContainerdService()
+			const (
+				// The unix socket for kubelet <-> containerdshim communication.
+				ep = "/var/run/containerdshim.sock"
+				// The unix socket for containerdshhim <-> containerd communication.
+				bindSocket = "/run/containerd/containerd.sock" // mikebrow get these from a config
+			)
+			kubeCfg.RemoteRuntimeEndpoint, kubeCfg.RemoteImageEndpoint = ep, ep
+
+			glog.V(2).Infof("Starting the GRPC client for containerd communication.")
+			// get the containerd client
+			dialOpts := []grpc.DialOption{grpc.WithInsecure(), grpc.WithTimeout(100 * time.Second)}
+			dialOpts = append(dialOpts,
+				grpc.WithDialer(func(addr string, timeout time.Duration) (net.Conn, error) {
+					return net.DialTimeout("unix", bindSocket, timeout)
+				},
+				))
```

Comment:
The indent here is not right. But this is POC. Let us fix this in the following PR.